admin/cron.php

```diff
@@ -62,21 +62,15 @@ function logs($s) {
 
 
 if(date('G') == 5) { // 5 hodin ráno
-  logs('Zálohuji databázi na FTP.');
-
-  if(!defined('FTP_ZALOHA_DB'))
-    throw new Exception('Není definována konstanta s adresou serveru pro zálohování.');
-
-  $backup = new Godric\DbBackup\DbBackup([
-    'sourceDb'  =>  [
-      'server'    =>  DB_SERV,
-      'user'      =>  DB_USER,
-      'password'  =>  DB_PASS,
-      'database'  =>  DB_NAME,
-    ],
-    'targetFtp' =>  FTP_ZALOHA_DB,
-  ]);
-  $backup->run();
+  logs('Zálohuji databázi.');
+
+  if(!defined('ZALOHA_DB_SLOZKA'))
+    throw new Exception('Není definována konstanta s adresářem pro zálohování.');
+
+  $db = dbConnect();
+  $dump = new MySQLDump($db);
+  $time = date('Y-m-d_His');
+  $dump->save(ZALOHA_DB_SLOZKA . "export_$time.sql.gz");
```

Comment:
Proměnné a konstanty s cestou prosím bez lomítek na konci. Když je to pokaždé jinak, tak z toho jsou pak ošklivé (byť stále funkční) cesty se dvěma lomítky. A taky se (mi) to hůře čte.

Navrhuji, aby tu bylo `$dump->save(ZALOHA_DB_SLOZKA . "/export_$time.sql.gz");`

Reply:
Akorát, že cesta se dvěma lomítky je na rozdíl od cesty s žádným lomítkem zcela funkční a bezpečná. Stejně tak, když nebude definovaná `ZALOHA_DB_SLOZKA`, tak se cesta místo poměrně bezpečného aktuálního adresáře změní na kořenovou složku.

Ale změním ti to, jestli se ti to tak líbí víc.

Reply:
Moc prosím, neprasme dvěma lomítky. To je peklo, které zažívám dnes a denně v práci jenom proto, že někdo nedokázal na začátku říct, jak to bude.

Velký problém s "už tam to lomítko přece je" je s tím, že dřív nebo později nebude. Proto jsou nakonec dvě lomítka radši všude možně. A tam kde nejsou nedokážeš na první pohled říct, zda to poslední je finální názve souboru, nebo slepenec

```php
$file = ZAKLADNI_CESTA . 'zaloha.sql'; // bude to /foo/bar/zaloha.sql nebo /foo/bar/nejaky_prefix_zaloha.sql ?
```

Reply:
Jo a taky dřív nebo později se ti ta dvě lomítka dostanou do URL - a tam už to poblém je, protože ani URL s lomítkem na konci a bez něj není pro většinou routerů stejná URL, natož se dvěma lomítky kdesi uprostřed.

Reply:
> Moc prosím, neprasme dvěma lomítky

Ale vždyť tam v žádné fázi dvě lomítka nebyla...

> A tam kde nejsou nedokážeš na první pohled říct, zda to poslední je finální názve souboru, nebo slepenec

Vidíš, a já bych to považoval za šikovnou funkci, že si můžu krom cesty nastavit i prefix souboru.

> Jo a taky dřív nebo později se ti ta dvě lomítka dostanou do URL - a tam už to problém je, protože ani URL s lomítkem na konci a bez něj není pro většinou routerů stejná URL, natož se dvěma lomítky kdesi uprostřed.

To by mne zajímalo, jaký máš testovací vzorek. Ano u prasáren typu WordPress nebo Drupal je koncové lomítko občas problém (pokud člověk správně nenastaví redirect) protože pak není poznat, jestli chceš přistupovat ke složce nebo stejně pojmenovanému souboru se správnou koncovkou; lomítka uprostřed drtivá většina rozumných webových serverů a proxy jednoduše vyhodí při normalizaci (jelikož se URI nakonec stejnak mapuje na souborový systém, tak nemá prázdná část cesty mezi dvěma lomítky moc význam). Ostatně viz: https://github.com////gamecon-cz//////gamecon/////////pull//////256////// nebo https://gamecon.cz//////novinky/////

Každopádně toto už je jen filozofická debata, to že ta, nebude více lomítek už si musíš vyřešit sám tím, že v produkčním nastavení budeš mít cestu bez finálního lomítka...

Reply:
> A tam kde nejsou nedokážeš na první pohled říct, zda to poslední je finální názve souboru, nebo slepenec

Mimochodem, stejný argument můžeš použít ale i opačně.
```php
ZAKLADNI_CESTA = "/cesta/k/souboru"
```

Bude to cesta ke konkrétnímu souboru `/cesta/k/souboru`, nebo to bude cesta k adresáři, kde se soubor nachází `/cesta/k/souboru/opravdovy_soubor` a nebo to bude nějaký slepenec `cesta/k/souboru.txt`? S lomítkem na konci je to jasný adresář 😄  Takže akorát volíš mezi tím, jestli chceš mít jasno v definici konstant nebo v pozdějším řetězení stringů.

PS: to že někde je `$file = CESTA . 'zaloha.sql';` ještě neznamená, že o pár řádků dál nemůže být ještě `$file = $file . "/neco";` takže stejně neznáš finální název souboru, dokud se to nepoužije v nějakém open/read/write volání nebo tak.

Reply:
Mám prostě špatnou zkušenost s více lomítky v cestě.

V URL je to špatné kvůli vyhledávačům a cachování, Symfony z toho udělá redirect, když je to v cestě souboru a pak se to někde objeví, byť jenom v logu, tak je to ošklivé.

A od začátku kódu Gameconu je trend bez lomítka na konci.

Ale rozhodně nejsme sami, kdo se nemůže shodnout https://stackoverflow.com/questions/980255/should-a-directory-path-variable-end-with-a-trailing-slash

Reply:
> Mám prostě špatnou zkušenost s více lomítky v cestě.
> 
> V URL je to špatné kvůli vyhledávačům a cachování: Symfony z toho udělá redirect. Když je to v cestě souboru a pak se to někde objeví, byť jenom v logu, tak je to ošklivé.

No to každopádně, z URL to prostě musí někde po cestě zmizet, buď to má mít člověk ošetřené na webovém serveru nějakým modulem pro normalizaci, nebo alespoň v aplikaci. Více různých URL pro stejnou stránku (obsah) je špatně.

> A od začátku kódu Gameconu je trend bez lomítka na konci.

To je taky dobře, také jsem zastáncem cest bez koncových lomítek (samozřejmě se stále bavíme jen o URL 😄 )

> Ale rozhodně nejsme sami, kdo se nemůže shodnout https://stackoverflow.com/questions/980255/should-a-directory-path-variable-end-with-a-trailing-slash

To určitě, každopádně z mé strany už nevím, co bych pro tebe mohl dalšího udělat (ten kód už je 13 dní dle tvé představy), já jen vysvětluju, že se toho není třeba bát.